dockerExtension.ts

```diff
@@ -157,13 +168,23 @@ export async function activate(ctx: vscode.ExtensionContext): Promise<void> {
         browseAzurePortal(context);
     });
 
-    ctx.subscriptions.push(vscode.debug.registerDebugConfigurationProvider('docker', new DockerDebugConfigProvider()));
+}
 
-    if (azureAccount) {
-        AzureUtilityManager.getInstance().setAccount(azureAccount);
-    }
+async function consolidateDefaultRegistrySettings(): Promise<void> {
+    const configOptions: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('docker');
+    let defaultRegistryPath = configOptions.get('defaultRegistryPath', '');
+    let defaultRegistry = configOptions.get('defaultRegistry', '');
 
-    activateLanguageClient(ctx);
+    if (defaultRegistry) {
+        if (defaultRegistryPath) { // combine both the settings, then notify the user
+            await configOptions.update('defaultRegistryPath', `${defaultRegistry}/${defaultRegistryPath}`, vscode.ConfigurationTarget.Workspace);
+        }
+        if (!defaultRegistryPath) {// assign defaultRegistry to defaultRegistryPath
+            await configOptions.update('defaultRegistryPath', `${defaultRegistry}`, vscode.ConfigurationTarget.Workspace);
+        }
+        await configOptions.update('defaultRegistry', '', false);
+        vscode.window.showInformationMessage("Moving forward, we shall deprecate the use of the defaultRegistry setting, combining it with defaultRegistryPath.")

```

Comment:
This implies you're *going* to do this, but you just did it.

`The 'docker.defaultRegistry' setting is now obsolete, and you should just use the 'docker.defaultRegistryPath' setting. Your settings have been updated to reflect this change.`